### ISSUE TITLE
Create ED2-RTM sub directory within ED2/

### ISSUE THREAD

**Issue by serbinsh** [EDmodel/ED2#63]
@rgknox 

@mdietze and I would like to add an additional directory within the main ED2 director to house (at least for now) our developing ED2-RTM module code base for our NASA TE project.  Right now this is the current structure:

```
../BRAMS
../cnt_lines.sh
../diff_version.sh
../ED
../EDTS
../Ramspost
../RAPP
../README.md
../R-utils
````

We would like to add a dir labeled "ED2-RTM" (or something similar if that may conflict with something else) where we develop the code needed to assimilate RS datasets within the ED2 RTM framework, "offline" from the full model.  For now, having the separate RTM code base within the main ED2 folder will help with compiling and linking since what we are developing uses existing ED2 libs and functions.  Would this be something we can do, at least for the short-term?  If it turns out to cause issues (which I don't expect) we could always move it.  While we are leveraging existing libs, we are not modifying the mainline.  

The other option is to create an additional repo within the EDmodel list of repos for this code.

Thanks!

**Comment by robkooper** [comment]
Why not start with a branch, that you push to. You can branch the main ED model, keep it in sync, but develop in your branch. Once you like it you can let other look at it as well and create a pull request at one point to merge it back in.

**Comment by mdietze** [comment]
@robkooper yes, that totally works. Shawn and I just wanted to check with @rgknox and others that everyone was cool with adding another top-level directory for these tools.  

Also, what Shawn means by "we're not modifying the mainline" is that this code will not change any of the code in the ED directory or any of the other high-level directories. We actually **do** want this code committed to the git mainline at some point since it will make it easier to keep this tool in sync with everything else and because we think this tool will be useful to others (ability to run the full radiative model in ED at a fine spectral resolution 'offline' from the full ED executable) for linking ED to all forms of optical remote sensing

**Comment by rgknox** [comment]
I am definitely in favor of adding a new folder with ED-RTM.  Thanks for
checking in on this Shawn.  Let me know if there is anything I can help
with.  I don't have opinion on whether you develop code as branch or simply
forking changes into main, I suppose I defer to Rob's suggestions, but
seeing how the code would have its own ED-RTM folder, there is no reason
not to develop in main from my perspective.

**Comment by serbinsh** [comment]
@Viskari 

At some point lets make sure this gets done.

**Comment by Viskari** [comment]
Sure, I have at the moment tried to get the initial phase running before
truly focusing on this.

**Comment by mdietze** [comment]
Would be good to get things up sooner rather than latter. @Viskari you're previous huge pull requests have been very problematic because it is virtually impossible to evaluate the code and provide constructive feedback with large pull requests. I would very much prefer to not repeat the difficulties we had with waiting until the two DART projects were working.

**Comment by rgknox** [comment]
Hey guys, I was curious about the scope of the ED-RTM work.  It seems to me this involves data assimilation and potentially new implementation of RT inside of ED?  On the latter I wanted to get a sense for what you are doing so I can make my efforts more complimentary if need be.

I have been working out a Perfect Plasticity Approximation transfer scheme in my "spare time", but have not put any code into the mainline yet. It may morph to incorporate some ideas by Nikolay Shabanov that include a spatial covariance matrix, but we will see. I have tested in Matlab only so far.  If others have similar interests or compimentary interests we should all coordinate.

Stochastic Modeling of Radiation Regime in Discontinuous Vegetation Canopies
Nikolay V. Shabanov,* Y. Knyazikhin,* Freˆdeˆric Baret,†and Ranga B. Myneni*

**Comment by mdietze** [comment]
Ryan, yes that's the basic idea -- the main focus is on the DA side of things, more than the alternative models, but we definitely want to evaluate structure as well. The design plan, which is based on coupling to the internal canopy model in ED2 rather than having to recode them externally, should allow any alternative ED2 canopy models to be incorporated within the data assimilation. 

Yes, we should definitely continue discussions on alternative canopy models!
